danielssonn/ethTO
# chore: add precommit hook

## Purpose

FYI I noticed we did not check in the `package-lock.json` files. These should be checked into the repo, so that we are all working with the same dependency trees. I removed this entry from `.gitignore`, so that they are included now (which is why they are in this PR). I would recommend running `npm ci` in `client` and `protocol` once this PR is merged, so that we are all working from a clean install from these `package-lock.json` files. Also, for the precommit hook to work, you either need to install dependencies (`npm i or npm ci`), or manually run the `prepare` script, e.g. `npm run prepare`.

1. Added `husky`, `lint-staged` and `prettier` in the `client` project
2. Added a few useful scripts
3. Added a precommit `husky` hook that will run the `lint-staged` script

## Testing Instructions

1. Pull down the branch, run `cd client && npm ci` and commit an edit, you should see `eslint` and/or `prettier` run (depending on the file extension(s) in the staged files)
2. Review code

## Checklist
